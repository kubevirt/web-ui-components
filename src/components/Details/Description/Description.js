import React from 'react';
import PropTypes from 'prop-types';

import { getDescription } from '../../../utils';
import { InlineEdit } from '../../InlineEdit';
import { Loading } from '../../Loading';
import { DASHES } from '../../../constants';

const descriptionFormFields = {
  description: {
    id: 'description-textarea',
    type: 'textarea',
  },
};

export const Description = ({ formValues, vm, editing, updating, LoadingComponent, onFormChange }) => {
  if (!(formValues && formValues.description)) {
    onFormChange({ value: getDescription(vm) }, 'description', true);
  }

  return (
    <InlineEdit
      formFields={descriptionFormFields}
      fieldsValues={formValues}
      editing={editing}
      updating={updating}
      LoadingComponent={LoadingComponent}
      onFormChange={onFormChange}
    >
      {getDescription(vm) || DASHES}
    </InlineEdit>
  );
};

Description.propTypes = {
  formValues: PropTypes.object,
  vm: PropTypes.object.isRequired,
  editing: PropTypes.bool,
  updating: PropTypes.bool,
  LoadingComponent: PropTypes.func,
  onFormChange: PropTypes.func,
};

Description.defaultProps = {
  formValues: undefined,
  editing: false,
  updating: false,
  LoadingComponent: Loading,
  onFormChange: () => {},
};
