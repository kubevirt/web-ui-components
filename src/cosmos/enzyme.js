import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
