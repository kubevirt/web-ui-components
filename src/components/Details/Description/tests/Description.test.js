import React from 'react';
import { shallow } from 'enzyme';

import { Description } from '..';

import { default as DescriptionFixture } from '../fixtures/Description.fixture';

const testDescription = () => <Description {...DescriptionFixture[0].props} />;

describe('<Description />', () => {
  it('renders correctly', () => {
    const component = shallow(testDescription());
    expect(component).toMatchSnapshot();
  });
});
