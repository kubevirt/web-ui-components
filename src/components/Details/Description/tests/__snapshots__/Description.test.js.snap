// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> renders correctly 1`] = `
<InlineEdit
  LoadingComponent={[Function]}
  editing={false}
  fieldsValues={
    Object {
      "description": Object {
        "value": "vm description",
      },
    }
  }
  formFields={
    Object {
      "description": Object {
        "id": "description-textarea",
        "type": "textarea",
      },
    }
  }
  id="description"
  onFormChange={[Function]}
  showLabels={false}
  updating={false}
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec lacus nibh, convallis vel nunc id,tempus vulputate augue. Proin eget nisl vel ante tincidunt accumsan vel at elit. Fusce eget tincidunt sem. Fusce cursus orci vitae nisl hendrerit mollis. Nullam at nulla ut ipsum malesuada laoreet a sit amet est.
</InlineEdit>
`;
